src/js/components/Form/Form.js

```diff
@@ -93,6 +93,12 @@ const Form = forwardRef(
           );
           setPendingValidation(undefined);
 
+          setRequiredFields(
```

Comment:
Just to be a bit more tight, can we parallel how we use setValidationResults()? For example:
`setRequiredFields(prevRequiredFields => prevRequiredFields.filter(n => Object.keys(validations.current).includes(n)));`

Reply:
ok